internal/io/csv.go

```diff
@@ -98,6 +101,37 @@ func ReadCSV(reader io.Reader, conf CSVConfig) (map[string]interface{}, []string
 		}
 	}
 
+	if conf.MissingColumnNameAlias != "" {
+		headerSet := strings.NewEmptyStringSet()
+		for count, h := range headers {
+
+			if h == "" {
+				h = conf.MissingColumnNameAlias
+			} else {
+				headerSet.Add(h)
+			}
+			headers[count] = h
+
+		}
+		// fmt.Println("Empty Column names have been alised ", conf.MissingColumnNameAlias)
+	}
+
+	if conf.AddColumnIndexToDuplicateColumns {
+
+		headerSet := strings.NewEmptyStringSet()
+		for count, h := range headers {
+
+			if headerSet.Contains(h) {
+				h = h + fmt.Sprint(count)
+			} else {
```

Comment:
I think this part potentially needs to be thought through a bit more to make it bullet proof wrt to duplicates. As is now you could construct header combinations that will fail the uniqueness constraints.
`COL,COL,COL1`, `COL,COL2,COL`

While probably not super common I'd prefer if we could handle this since its a library and you don't really know what users will use it for. We could come up with a naming scheme like <NAME><COL>.<ENUMERATOR> where the ENUMERATOR is just added in case there are conflicts and is incremented until it does not conflict with an already existing column name.

It would be great to add a couple of unit tests for these corner cases.